### ISSUE TITLE
SSO does not honor custom ca-bundle configuration

### ISSUE THREAD

**Issue by kjluedke** [aws/aws-cli#7602]
### Describe the bug

Executing an sso subcommand does not honor the --ca-bundle option or ca-bundle configuration.  The value provided to the CLI is ignored in favor of the default certificate bundle.

### Expected Behavior

I expected my specified bundle to be used.

### Current Behavior

The default bundle at `C:\Program Files\Amazon\AWSCLIV2\awscli\botocore\cacert.pem` is used.

### Reproduction Steps

Execute the following command:

`aws sso login --ca-bundle C:\Users\myUser\aws-ca-oidc-bundle.pem --debug`

Observe the following line in the debug output:
`2023-01-17 09:53:20,292 - MainThread - botocore.httpsession - DEBUG - Certificate path: C:\Program Files\Amazon\AWSCLIV2\awscli\botocore\cacert.pem`

### Possible Solution

Our temporary workaround has been to replace the contents of `C:\Program Files\Amazon\AWSCLIV2\awscli\botocore\cacert.pem` with our custom bundle.

### Additional Information/Context

This defect might be related to https://github.com/aws/aws-cdk/issues/21328 

### CLI version used

aws-cli/2.9.15 Python/3.9.11

### Environment details (OS name and version, etc.)

Windows/10 exe/AMD64

**Comment by tim-finnigan** [comment]
Thanks @kjluedke for reporting this. The team confirmed the issue and will need to investigate it further. We will share any updates here.

**Comment by esmcelroy** [comment]
Reporting that I'm having the same issue. Version: aws-cli/2.11.0 Python/3.11.2 Windows/10 exe/AMD64
Pulling the bottom-most traceback after running `aws configure sso --no-verify-ssl --debug`:
```
Traceback (most recent call last):
  File "awscli\clidriver.py", line 460, in main
  File "awscli\customizations\commands.py", line 151, in __call__
  File "awscli\customizations\commands.py", line 205, in __call__
  File "awscli\customizations\configure\sso.py", line 516, in _run_main
  File "awscli\customizations\sso\utils.py", line 72, in do_sso_login
  File "awscli\botocore\utils.py", line 3052, in fetch_token
  File "awscli\botocore\utils.py", line 3037, in _token
  File "awscli\botocore\utils.py", line 2947, in _poll_for_token
  File "awscli\botocore\utils.py", line 2918, in _registration
  File "awscli\botocore\utils.py", line 2875, in _register_client
  File "awscli\botocore\client.py", line 341, in _api_call
  File "awscli\botocore\client.py", line 683, in _make_api_call
  File "awscli\botocore\client.py", line 703, in _make_request
  File "awscli\botocore\endpoint.py", line 101, in make_request
  File "awscli\botocore\endpoint.py", line 155, in _send_request
  File "awscli\botocore\endpoint.py", line 199, in _do_get_response
  File "awscli\botocore\endpoint.py", line 271, in _send
  File "awscli\botocore\httpsession.py", line 475, in send
botocore.exceptions.SSLError: SSL validation failed for https://oidc.us-west-2.amazonaws.com/client/register [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:992)
```

Which suggests that at no point in the creation of the SSO request session is the session made aware of the verify global variable. Throughout the various customization files, we can see different sessions being created with different contexts (and usually with reference to the verify value):
For configservice.subscribe: https://github.com/aws/aws-cli/blob/2.11.0/awscli/customizations/configservice/subscribe.py#L120-L130
For codedeploy.register: https://github.com/aws/aws-cli/blob/2.11.0/awscli/customizations/codedeploy/register.py#L76-L81

Then, for configure.sso: https://github.com/aws/aws-cli/blob/2.11.0/awscli/customizations/configure/sso.py#L528-L529

This may not be the issue, but it's clear that the verify option isn't being correctly parsed during token-get operations, sso configure operations and other SSO-based operations. It's possible the issue is more widespread, but currently it's a clear issue with SSO. We've also worked around this issue by updating the cacerts.pem bundle, but this isn't a sustainable or maintainable fix.

**Comment by mrkharvi01** [comment]
I have found the issue and I would like to work on this. I will submit the PR later

**Comment by jboeshart** [comment]
Still seeing this as an issue as well. Running `aws-cli/2.13.0 Python/3.11.4 Windows/10 exe/AMD64 prompt/off`. Since this has been an issue since at least January, any chance the CLI documentation could be updated to reflect that neither the config file profile setting nor command line parameter work in the latest version? It does seem to work when specifying through environment variable though, both in Windows CMD and Windows PowerShell.

**Comment by jboeshart** [comment]
Also looks like this was reported here https://github.com/aws/aws-cli/issues/7552 as well.

**Comment by jboeshart** [comment]
I just upgraded to 2.13.34 and it seems to be honoring the `ca_bundle` configuration file setting, haven't tried the other options. I don't see anything in the release notes that indicates there was a fix released, so I'm not sure if it's fully fixed or not.

**Comment by jboeshart** [comment]
Strike that, looks like the certs must have been cached or something, as it's no longer honoring the config file and I've had to manually add the cert of our forward proxy to "C:\Program Files\Amazon\AWSCLIV2\awscli\botocore\cacert.pem" in order for it to work. Underlying issue still looks to be there.

**Comment by esmcelroy** [comment]
I was gonna say, I didn't see any code changes in the last ~year to the SSO functions, aside from a docs update! 

If I can assemble myself a testing environment with a self-signed certificate injection, I'll put together the fix - it shouldn't be too bad... It looks like when it creates the session it's just not receiving the parsed globals.

**Comment by Farmbuyer** [comment]
Any progress on this by chance?  We're having to use workarounds here for overly intrusive SSL inspection; until the IT department implements the proper fixes documented elsewhere, the individual programmers need to use `--no-verify-ssl` for all commands.

Unfortunately, when SSO ignores it, there's no way to make progress.

**Comment by joaocc** [comment]
Hi. Is there any progress on this? aws-cli/2.15.18 still exhibits this behaviour. Thanks

**Comment by lambda-science** [comment]
Any update ? Facing the same issue with Zscaler, it's terrible

**Comment by couryrr** [comment]
I am unable to run `aws configure sso` as well with the cli. I am getting something along the lines of 

> SSL validation failed for <uri> [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)

If I run:

> curl -v <start_url>

In that output I can see that this pem is being used: `CAfile: /etc/ssl/cert.pem`

Which it does have Zscaler entry: `[CN=Zscaler Intermediate Root CA](issuer: C=US; ST=California; O=Zscaler Inc.; OU=Zscaler Inc.; CN=Zscaler Intermediate Root CA (<removed>) (t))`

when I run `aws configure sso --ca-bundle /etc/ssl/cert.pem` or export the env I still get an issue with the SSL validation.

Even bypassing with `--no-verify-ssl` fails.

**Comment by esmcelroy** [comment]
> I am unable to run `aws configure sso` as well with the cli. I am getting something along the lines of
> 
> > SSL validation failed for  [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
> 
> If I run:
> 
> > curl -v <start_url>
> 
> In that output I can see that this pem is being used: `CAfile: /etc/ssl/cert.pem`
> 
> Which it does have Zscaler entry: `[CN=Zscaler Intermediate Root CA](issuer: C=US; ST=California; O=Zscaler Inc.; OU=Zscaler Inc.; CN=Zscaler Intermediate Root CA (<removed>) (t))`
> 
> when I run `aws configure sso --ca-bundle /etc/ssl/cert.pem` or export the env I still get an issue with the SSL validation.
> 
> Even bypassing with `--no-verify-ssl` fails.

If you're able to build a fresh version of the CLI from source, you could try patching in my fix posted here: https://github.com/aws/aws-cli/issues/7602#issuecomment-1881710592

I haven't had access to an environment in which to test this fix since I wrote it - may need to be updated, as it was specific to 2.15.7 at the time.

**Comment by esmcelroy** [comment]
the line in question is here: https://github.com/aws/aws-cli/blob/37a410088eb035d52da38545f1c8b9330b0a95e7/awscli/customizations/configure/sso.py#L529

**Comment by esmcelroy** [comment]
in almost every other module I could find, the `self._session.create_client` call included the `verify=parsed_globals.verify_ssl` argument. I think it's just an oversight that it's not included here...

**Comment by couryrr** [comment]
> > I am unable to run `aws configure sso` as well with the cli. I am getting something along the lines of
> > > SSL validation failed for  [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
> > 
> > 
> > If I run:
> > > curl -v <start_url>
> > 
> > 
> > In that output I can see that this pem is being used: `CAfile: /etc/ssl/cert.pem`
> > Which it does have Zscaler entry: `[CN=Zscaler Intermediate Root CA](issuer: C=US; ST=California; O=Zscaler Inc.; OU=Zscaler Inc.; CN=Zscaler Intermediate Root CA (<removed>) (t))`
> > when I run `aws configure sso --ca-bundle /etc/ssl/cert.pem` or export the env I still get an issue with the SSL validation.
> > Even bypassing with `--no-verify-ssl` fails.
> 
> If you're able to build a fresh version of the CLI from source, you could try patching in my fix posted here: [#7602 (comment)](https://github.com/aws/aws-cli/issues/7602#issuecomment-1881710592)
> 
> I haven't had access to an environment in which to test this fix since I wrote it - may need to be updated, as it was specific to 2.15.7 at the time.

After a couple of attempts I am a bit stumped on an error that the build is throwing. I am on a Mac M2.

**Steps:**

- cloned this repo.
- git switch v2
- python3 -m vene .
- source bin/activate
- ./configure --with-download-deps 
- make

At this point it fails with:

> note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for ruamel-yaml-clib
Failed to build ruamel-yaml-clib
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (ruamel-yaml-clib)
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/__main__.py", line 125, in <module>
    main()
    ~~~~^^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/__main__.py", line 121, in main
    parsed_args.func(parsed_args)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/__main__.py", line 39, in build
    aws_venv = _bootstap_venv(
        parsed_args.build_dir,
        parsed_args.artifact,
        parsed_args.download_deps,
    )
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/__main__.py", line 74, in _bootstap_venv
    aws_venv.bootstrap(artifact_type, download_deps)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/awscli_venv.py", line 58, in bootstrap
    self._install_requirements(SYSTEM_SANDBOX_REQUIREMENTS_LOCK)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/awscli_venv.py", line 79, in _install_requirements
    self._pip_install(
    ~~~~~~~~~~~~~~~~~^
        ["--no-build-isolation", "-r", requirements_file],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        cwd=cwd,
        ^^^^^^^^
    )
    ^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/awscli_venv.py", line 135, in _pip_install
    self._utils.run(args, **run_kwargs)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/Users/coury.richards/Documents/code/open-source/aws-cli/./backends/build_system/utils.py", line 235, in run
    return subprocess.run(args, **kwargs)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/Users/coury.richards/Documents/code/open-source/aws-cli/build/venv/bin/python', '-m', 'pip', 'install', '--no-build-isolation', '-r', PosixPath('/Users/coury.richards/Documents/code/open-source/aws-cli/requirements/download-deps/system-sandbox-lock.txt')]' returned non-zero exit status 1.
make: *** [build] Error 1

**Comment by esmcelroy** [comment]
oh! IIRC the instructions I followed to build the CLI were the ones found here: https://github.com/aws/aws-cli/blob/develop/CONTRIBUTING.md#cli-development-version

**Comment by couryrr** [comment]
Oh what an oversight on my part. I was too busy reading the v2 code I did not look at the default branch for that markdown. Thanks!

I was able to build on develop just fine which gives me the v1 aws-cli.

I moved over to v2 and ran:

- python3.12 -m venv .
- source bin/activate
- python3.12 -m pip --no-cache install -r requirements.txt
- python3.12 -m pip install -e .

This is currently failing:

```
Building wheels for collected packages: awscli
  Building editable for awscli (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building editable for awscli (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [26 lines of output]
      Injecting script tox -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/tox
      Injecting script pyi-grab_version -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/pyi-grab_version
      Injecting script pyi-makespec -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/pyi-makespec
      Injecting script Activate.ps1 -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/Activate.ps1
      Injecting script aws.cmd -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/aws.cmd
      Injecting script python3 -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/python3
      Traceback (most recent call last):
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 76, in build_editable
          _inject_editable_wheel_extras(os.path.join(wheel_directory, whl_filename))
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 195, in _inject_editable_wheel_extras
          _inject_scripts(extracted_wheel_dir)
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 267, in _inject_scripts
          _rewrite_shebang(target)
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 179, in _rewrite_shebang
          lines = f.read().split("\n")
                  ^^^^^^^^
        File "<frozen codecs>", line 322, in decode
      UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for awscli
Failed to build awscli
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (awscli)
```

**Comment by couryrr** [comment]
I have attempted this install (several times) on the `v2` branch and cannot get the final step to run. I do not do a lot of python development so this is outside my wheelhouse at this point. Any suggestion to get a successful build I am more than happy to try.

Running
- `python3 --version`
Python 3.12.7

- `python3 -m venv .`
succeeds 

- `. ./bin/activate`
activates virtual environment

- `python --version`
Python 3.12.7
This does not exist outside the virtual env

- `python -m pip install -r requirements.txt` 
succeeds (a lot of output for the downloads)

- `python -m pip install -e .`
Fails:

>...
Building wheels for collected packages: awscli
  Building editable for awscli (pyproject.toml) ... error
  error: subprocess-exited-with-error
  × Building editable for awscli (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [26 lines of output]
      Injecting script tox -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/tox
      Injecting script pyi-grab_version -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/pyi-grab_version
      Injecting script pyi-makespec -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/pyi-makespec
      Injecting script Activate.ps1 -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/Activate.ps1
      Injecting script aws.cmd -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/aws.cmd
      Injecting script python3 -> /Users/coury.richards/Documents/code/open-source/aws-cli/build/unpacked_wheel/awscli-2.22.4.data/scripts/python3
      Traceback (most recent call last):
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 76, in build_editable
          _inject_editable_wheel_extras(os.path.join(wheel_directory, whl_filename))
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 195, in _inject_editable_wheel_extras
          _inject_scripts(extracted_wheel_dir)
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 267, in _inject_scripts
          _rewrite_shebang(target)
        File "/Users/coury.richards/Documents/code/open-source/aws-cli/backends/pep517.py", line 179, in _rewrite_shebang
          lines = f.read().split("\n")
                  ^^^^^^^^
        File "<frozen codecs>", line 322, in decode
      UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte
      [end of output]
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for awscli
Failed to build awscli
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (awscli)

**Comment by couryrr** [comment]
Well this has been a learning experience for python development. I was setting the `python3 -m venv .` which fails as mentioned above. I changed it to `python3 -m venv .venv && source .venv/bin/activate` and I can build!

With that complete inside of the virtualenv I can follow these [docs](https://github.com/aws/aws-cli/tree/v2?tab=readme-ov-file#cli-dev-version) for v2.

I have the following for aws
```
~/Documents/code/open-source/aws-cli (v2*) » which aws                                                                                                                                                 252 ↵ coury.richards@AFSMW697035304
/Users/coury.richards/Documents/code/open-source/aws-cli/.venv/bin/aws

~/Documents/code/open-source/aws-cli (v2*) » aws --version                                                                                                                                             130 ↵ coury.richards@AFSMW697035304
aws-cli/2.22.4 Python/3.9.6 Darwin/23.6.0 source/arm64
```

From there I tested the base case: 
- no code changes
- no ssl settings

The expectation was a failure with ssl validation. 

The outcome was a successful run of configure sso. The dev build appears to point the cert path here: 

`2024-11-25 13:37:37,185 - MainThread - botocore.httpsession - DEBUG - Certificate path: /Users/coury.richards/Documents/code/open-source/aws-cli/awscli/botocore/cacert.pem`

I made the update to the line of code

```
diff --git a/awscli/customizations/configure/sso.py b/awscli/customizations/configure/sso.py
index 7b4fbe230..ceaf13fe6 100644
--- a/awscli/customizations/configure/sso.py
+++ b/awscli/customizations/configure/sso.py
@@ -526,7 +526,7 @@ class ConfigureSSOCommand(BaseSSOConfigurationCommand):
             signature_version=UNSIGNED,
             region_name=sso_registration_args['sso_region'],
         )
-        sso = self._session.create_client('sso', config=client_config)
+        sso = self._session.create_client('sso', config=client_config, verify=parsed_globals.verify_ssl)

         sso_account_id, sso_role_name = self._prompt_for_sso_account_and_role(
             sso, sso_token
(END)
```

Ran the command: `aws configure sso --debug --ca-bundle /etc/ssl/cert.pem`

The expectation now was that the debug line would change. It did not.

I cannot confirm that this issue is fixed with the code changes. It also appears that the dev build behave differently from the version I have on my machine.

**Comment by esmcelroy** [comment]
interesting! My suggestion (and IIRC how I built it when I was testing) is to check out the tagged ref of a released version and build from that tag.

Weird that it worked without any issues in the venv... Did you try again with your installed copy after it worked?

**Comment by ashovlin** [comment]
Hey all, as @esmcelroy pointed out above, we are missing the the `verify` argument when calling `create_client` in the SSO code paths. This is why `--ca-bundle` isn't honored. 

The [`AWS_CA_BUNDLE` environment variable](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list) and [`ca_bundle` config setting](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-config-ca_bundle) should still work for now. Let us know if they aren't. 
```
% AWS_CA_BUNDLE=~/path/to/certificate.pem aws sso login --profile <profile>
```

This came up again internally too recently, I'll bring up in our next grooming again.

**Comment by couryrr** [comment]
> Hey all, as @esmcelroy pointed out above, we are missing the the `verify` argument when calling `create_client` in the SSO code paths. This is why `--ca-bundle` isn't honored.
> 
> The [`AWS_CA_BUNDLE` environment variable](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list) and [`ca_bundle` config setting](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-config-ca_bundle) should still work for now. Let us know if they aren't.
> 
> ```
> % AWS_CA_BUNDLE=~/path/to/certificate.pem aws sso login --profile <profile>
> ```
> 
> This came up again internally too recently, I'll bring up in our next grooming again.

This does not seem to have any impact on the ssl validation. I am still failing.

**Comment by couryrr** [comment]
After some additional debugging with the networking folk from my side. We tried a new policy and a different cert bundle and `aws sso` is correctly working with a specified bundle. I can use the env on my system, passing it in as the above or using the `--ca-bundle` flag all successfully now.

It would appear that this is more a network config issue then an aws-cli problem. I cannot speak to the specifics of what was changed but I would work more closely with whomever has SSL inspection setup.
